### PR TITLE
docs: token classification tutorial

### LINKED ISSUE #5173: [DOCS] Token classification tutorial

### PR DESCRIPTION
**Pull request by sdiazlor**
# Pull Request Template

Closes #5173 

**Type of change**

- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation

### CODE REVIEW

**Comment by sdiazlor** [comment]
@davidberenstein1957 After querying a dataset, the best way to iterate over the retrieved records is using .to_list. I think it's useful to change it in the how-to guides instead of the current list(). WDYT?

**Comment by davidberenstein1957** [comment]
@sdiazlor I see there is something wrong with the doc workflow. I will check what is going wrong here. I think it might be the GH actions extension we are using because the message seems to have been printed correctly within the actions workflow.

**Comment by davidberenstein1957** [comment]
@sdiazlor, thanks for handling this. I think it will offer a good basis for people to get started.

Some high level comments, will potentially leave some more in the notebook :) 
- not 100% sure what you mean by `list` vs `to_list`. Ideally, we should unify communication on how we intend the API to be used. In that case the `Records.to_list` would be the preferred way and we should try to use this in the rest of the guides too.
- I think we should try to format cells in the notebook to adhere to line width etc, this can be done by right-clicking on a notebook and then selecting "format". Perhaps you can add `ipynb` files to the ruff pre.commit formatter?
- I think we can remove "Task" from the naming of both the tutorials.

File: argilla/docs/tutorials/token_classification.ipynb [hunk 1]
```diff
@@ -0,0 +1,705 @@
+{
```
Comment (davidberenstein1957) [inline argilla/docs/tutorials/token_classification.ipynb:1]: <p>should we emphasize something about Span Classification and overlapping entiites when defining the question?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/argilla-io/argilla/pull/5183/discussion/#comment-1670148667'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='14'/>

Comment (davidberenstein1957) [inline argilla/docs/tutorials/token_classification.ipynb:1]: <p>maybe the phrasing is a bit weird. "We would need to add data" sound more intuitive.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/argilla-io/argilla/pull/5183/discussion/#comment-1670148668'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='18'/>

Comment (davidberenstein1957) [inline argilla/docs/tutorials/token_classification.ipynb:1]: <p>maybe explain why we are adding suggestions_</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/argilla-io/argilla/pull/5183/discussion/#comment-1670148669'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='23'/>

**Comment by sdiazlor** [comment]
@davidberenstein1957 Thanks for your comments!

I meant that when doing this, it is difficult to iterate over the records and retrieve the needed data(fields, suggestions, responses, etc.).
`filtered_records = list(dataset.records(status_filter))`

Instead, as in the tutorials, it's more helpful to use.
`filtered_records = dataset.records(status_filter).to_list(flatten=True)`

But your answer solved the doubt, so I'll update it.

Thank you for the formatting reminder, ipynb files were not included.

**Comment by sdiazlor** [comment]
@davidberenstein1957 I applied the changes. However, it doesn't let me update the length too much, especially for the scripts due to pre-commit. We should update the length in pyproject.toml from 120 to the default 88.

**Comment by davidberenstein1957** [comment]
@sdiazlor, feel free to change it to `line-width=88` in another PR :) 

https://github.com/argilla-io/argilla/issues/5211

File: argilla/mkdocs.yml [hunk 2]
```diff
@@ -145,6 +145,7 @@ nav:
   - Tutorials:
       - tutorials/index.md
       - Text classification task: tutorials/text_classification.ipynb
+      - Token classification task: tutorials/token_classification.ipynb
```
Comment (davidberenstein1957) [inline argilla/mkdocs.yml]:
```suggestion
      - Text classification: tutorials/text_classification.ipynb
      - Token classification: tutorials/token_classification.ipynb
```

File: argilla/docs/tutorials/index.md [hunk 3]
```diff
@@ -17,4 +17,11 @@ These are the tutorials for *the Argilla SDK*. They provide step-by-step instruc
     Learn about a standard workflow to improve data quality for a text classification task.
     [:octicons-arrow-right-24: Tutorial](text_classification.ipynb)
 
-</div>
\ No newline at end of file
+-   __Token classification task__
```
Comment (davidberenstein1957) [inline argilla/docs/tutorials/index.md]:
```suggestion
-   __Token classification__
```

**Review by davidberenstein1957 — APPROVED** [review]

**Pull request merged**